improver_tests/utilities/test_cube_checker.py

```diff
@@ -97,69 +102,70 @@ class Test_check_cube_coordinates(IrisTest):
     dimension coordinates in a new cube if they were dimension coordinates in
     the progenitor cube."""
 
+    def setUp(self):
+        """Set up a cube."""
+        data = np.ones((1, 16, 16), dtype=np.float32)
+        data[:, 7, 7] = 0.0
+        self.cube = set_up_variable_cube(
+            data, "precipitation_amount", "kg m^-2", "equalarea",
+        )
+
     def test_basic(self):
         """Test returns iris.cube.Cube."""
-        cube = set_up_cube()
-        result = check_cube_coordinates(cube, cube)
+        result = check_cube_coordinates(self.cube, self.cube)
         self.assertIsInstance(result, Cube)
 
     def test_basic_transpose(self):
         """Test when we only want to transpose the new_cube."""
-        cube = set_up_cube()
-        new_cube = set_up_cube()
-        new_cube.transpose([3, 2, 0, 1])
-        result = check_cube_coordinates(cube, new_cube)
-        self.assertEqual(result.dim_coords, cube.dim_coords)
+        new_cube = self.cube.copy()
+        new_cube.transpose([2, 1, 0])
+        result = check_cube_coordinates(self.cube, new_cube)
+        self.assertEqual(result.dim_coords, self.cube.dim_coords)
 
     def test_coord_promotion(self):
         """Test that scalar coordinates in new_cube are promoted to dimension
         coordinates to match the parent cube."""
-        cube = set_up_cube()
-        new_cube = iris.util.squeeze(cube)
-        result = check_cube_coordinates(cube, new_cube)
-        self.assertEqual(result.dim_coords, cube.dim_coords)
+        new_cube = iris.util.squeeze(self.cube)
+        result = check_cube_coordinates(self.cube, new_cube)
+        self.assertEqual(result.dim_coords, self.cube.dim_coords)
 
     def test_coord_promotion_only_dim_coords_in_parent(self):
         """Test that only dimension coordinates in the parent cube are matched
         when promoting the scalar coordinates in new_cube. Here realization is
         made into a scalar coordinate on the parent, and so should remain a
         scalar in new_cube as well."""
-        cube = set_up_cube()
-        new_cube = iris.util.squeeze(cube)
-        cube = cube[0]
+        new_cube = iris.util.squeeze(self.cube)
+        cube = self.cube[0]
         result = check_cube_coordinates(cube, new_cube)
         self.assertEqual(result.dim_coords, cube.dim_coords)
```

Comment:
This could be done in a single test now that `set_up_variable_cube` is being used, since the cube that's created has existing scalar coordinates (time, forecast reference time and forecast period).  So `test_coord_promotion` already covers everything in the second test; or you could add an explicit statement to the first test if you wanted to check that the scalar non-dimensional coordinates were preserved.

Reply:
Removed second test and added a non-dim coords check to the first.